### PR TITLE
chore: Update KMEA KMManager for 14.0

### PR DESCRIPTION
**Pull request by darcywong00**
This starts to address #180

Add new KMManager API's
* getDefaultKeyboard()
* setDefaultKeyboard()
* getMaySendCrashReport() (See keymanapp/keyman#3825)
* setMaySendCrashReport()

Modifications
* Update the add/get/set Keyboard methods to use the new Keyboard type. (Mark HashMap dictionaries as *(Deprecated)*)

### CODE REVIEW

**Review by mcdurdin — CHANGES_REQUESTED** [review]
A few changes in style requested

File: developer/engine/android/14.0/KMManager/addKeyboard.php [hunk 1]
```diff
@@ -30,18 +50,30 @@
...
-    kbInfo.put(KMManager.KMKey_Font, "aava1.ttf");
+    Keyboard kbInfo = new Keyboard(
+      "basic_kbdtam99", // Package ID - filename of the .kmp file
+      "basic_kbdtam99", // Keyboard ID
+      "Tamil 99 Basic", // Keyboard Name
+      "ta",             // Language ID
+      "Tamil",          // Language Name
+      "1.0",            // Keyboard Version
+      null,             // URL to help documentation if available
+      "",               // URL to latest .kmp file
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/engine/android/14.0/KMManager/addKeyboard.php:68]: Either `null` or `""` work for those fields.

If the help URL is not a welcome.htm file, the Keyboard constructor will make the help link https://help.keyman.com/keyboard/<keyboardID>

If the latest kmp link is null or empty, a resource update is not available.

File: developer/engine/android/14.0/KMManager/getCurrentKeyboardInfo.php [hunk 2]
```diff
@@ -22,23 +22,44 @@
...
 <h3 id="Returns" name="Returns">Returns</h3>
-<p>Returns an information dictionary of the current keyboard with keys and values defined as <code>HashMap&lt;String key, String value&gt;</code>.</p>
+<p>Returns the current keyboard information of <code>Keyboard</code> type.</p>
+
+<h2 id="Description" name="Description">Description</h2>
+<p>Use this method to get details of the currently selected keyboard. Details include package ID, keyboard ID, language ID, keyboard name, language name and fonts.</p>
+
+<br><hr>
+
+<h2 id="Syntax" name="Syntax">Syntax (Deprecated)</h2>
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply developer/engine/android/14.0/KMManager/getCurrentKeyboardInfo.php:32]: OK. KMManager wasn't noting history, so I added those section

File: developer/engine/android/14.0/KMManager/addKeyboard.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/engine/android/14.0/KMManager/addKeyboard.php:68]: Okay, for the example can we use `null` please?

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**